attest/verifier/types/src/convert/collateral.rs

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! Conversions from prost message types into common crate rust types.
+
+use crate::{prost, ConversionError};
+use alloc::{
+    string::{String, ToString},
+    vec::Vec,
+};
+use mc_sgx_dcap_sys_types::sgx_ql_qve_collateral_t;
+use mc_sgx_dcap_types::Collateral;
+use x509_cert::{
+    der::{pem::LineEnding, Decode, Encode, EncodePem},
+    Certificate,
+};
+
+impl TryFrom<prost::Collateral> for Collateral {
+    type Error = ConversionError;
+
+    fn try_from(value: prost::Collateral) -> Result<Self, Self::Error> {
+        // Note: sgx_collateral uses null bytes at the end of most arrays.
+
+        let mut sgx_collateral = version_3_1_empty_collateral();
```

Comment:
I'm not fond of the low level building up of the collateral via the `sgx_ql_qve_collateral_t`. I think if this lived in `mc-sgx-dcap-types` it wouldn't be as questionable. At the same time I think building up the `Collateral` via a normal constructor maybe error prone as all 7 of the members should exist to be valid.

As the commit message says It felt like doing the prost types and conversion here in the main MobileCoin repo feels like the quickest way to get it done, while creating a binary representation that should be stable and ergonomic for the future.